diff --git a/src/core/bittorrent/session.cpp b/src/core/bittorrent/session.cpp
index d27d9e2..2c473ef 100644
--- a/src/core/bittorrent/session.cpp
+++ b/src/core/bittorrent/session.cpp
@@ -87,7 +87,8 @@ using namespace BitTorrent;
 #include "torrenthandle.h"
 #include "session.h"
 
-static const char PEER_ID[] = "qB";
+//static const char PEER_ID[] = "qB";
+static const char PEER_ID[] = "UT";
 static const char RESUME_FOLDER[] = "ResumeData";
 static const int MAX_TRACKER_ERRORS = 2;
 
@@ -161,7 +162,8 @@ Session::Session(QObject *parent)
     // Creating BitTorrent session
 
     // Construct session
-    libt::fingerprint fingerprint(PEER_ID, VERSION_MAJOR, VERSION_MINOR, VERSION_BUGFIX, VERSION_BUILD);
+    //libt::fingerprint fingerprint(PEER_ID, VERSION_MAJOR, VERSION_MINOR, VERSION_BUGFIX, VERSION_BUILD);
+    libt::fingerprint fingerprint(PEER_ID, 3, 3, 0, 0);
     m_nativeSession = new libt::session(fingerprint, 0);
     Logger::instance()->addMessage(tr("Peer ID: ") + Utils::String::fromStdString(fingerprint.to_string()));
 
@@ -362,7 +364,8 @@ void Session::setSessionSettings()
     Logger* const logger = Logger::instance();
 
     libt::session_settings sessionSettings = m_nativeSession->settings();
-    sessionSettings.user_agent = "qBittorrent " VERSION;
+    //sessionSettings.user_agent = "qBittorrent " VERSION;
+    sessionSettings.user_agent = "uTorrent/3300(29667)";
     //std::cout << "HTTP User-Agent is " << sessionSettings.user_agent << std::endl;
     logger->addMessage(tr("HTTP User-Agent is %1").arg(Utils::String::fromStdString(sessionSettings.user_agent)));
 
