diff -U 3 -dHrN -- astyle.orig/src/ASFormatter.cpp astyle/src/ASFormatter.cpp
--- astyle.orig/src/ASFormatter.cpp	2014-12-11 22:42:25.000000000 +0800
+++ astyle/src/ASFormatter.cpp	2015-05-21 22:29:18.184543791 +0800
@@ -1517,6 +1517,11 @@
 		if (currentChar == ';'
 		        || (currentChar == ',' && shouldPadOperators))
 		{
+			// Remove spaces before the ','
+			const size_t len = formattedLine.length();
+			size_t lastText = formattedLine.find_last_not_of(' ');
+			if (lastText != string::npos && lastText < len - 1)
+				formattedLine.resize(lastText + 1);
 			char nextChar = ' ';
 			if (charNum + 1 < (int) currentLine.length())
 				nextChar = currentLine[charNum + 1];
