diff -uprN astyle.orig/src/ASFormatter.cpp astyle/src/ASFormatter.cpp
--- astyle.orig/src/ASFormatter.cpp	2013-11-02 04:31:48.000000000 +0800
+++ astyle/src/ASFormatter.cpp	2014-11-16 19:42:17.401436109 +0800
@@ -1775,6 +1775,16 @@ void ASFormatter::setAttachClass(bool st
 }
 
 /**
+ * set option to break struct brackets
+ *
+ * @param state        true = break, false = use style default.
+ */
+void ASFormatter::setBreakStruct(bool state)
+{
+	shouldBreakStruct = state;
+}
+
+/**
  * set option to attach extern "C" brackets
  *
  * @param state        true = attach, false = use style default.
@@ -4664,6 +4674,11 @@ bool ASFormatter::isCurrentBracketBroken
 	{
 		return false;
 	}
+	else if (shouldBreakStruct
+	         && isBracketType((*bracketTypeStack)[stackEnd], STRUCT_TYPE))
+	{
+		return true;
+	}
 	else if (shouldAttachInline
 	         && isCStyle()			// for C++ only
 	         && bracketFormatMode != RUN_IN_MODE
diff -uprN astyle.orig/src/astyle.h astyle/src/astyle.h
--- astyle.orig/src/astyle.h	2013-11-02 04:31:48.000000000 +0800
+++ astyle/src/astyle.h	2014-11-16 19:40:22.081592376 +0800
@@ -652,6 +652,7 @@ class ASFormatter : public ASBeautifier
 		void setAttachExternC(bool state);
 		void setAttachNamespace(bool state);
 		void setAttachInline(bool state);
+		void setBreakStruct(bool state);
 		void setBracketFormatMode(BracketMode mode);
 		void setBreakAfterMode(bool state);
 		void setBreakClosingHeaderBracketsMode(bool state);
@@ -863,6 +864,7 @@ class ASFormatter : public ASBeautifier
 		bool shouldAttachNamespace;
 		bool shouldAttachClass;
 		bool shouldAttachInline;
+		bool shouldBreakStruct;
 		bool isInLineComment;
 		bool isInComment;
 		bool isInCommentStartLine;
diff -uprN astyle.orig/src/astyle_main.cpp astyle/src/astyle_main.cpp
--- astyle.orig/src/astyle_main.cpp	2013-11-02 04:31:48.000000000 +0800
+++ astyle/src/astyle_main.cpp	2014-11-16 19:41:04.488201578 +0800
@@ -1441,6 +1441,9 @@ void ASConsole::printHelp() const
 	(*_err) << endl;
 	(*_err) << "Bracket Modify Options:\n";
 	(*_err) << "-------------------------\n";
+	(*_err) << "    --break-structs  OR  -xs\n";
+	(*_err) << "    Break brackets from a struct statement.\n";
+	(*_err) << endl;
 	(*_err) << "    --attach-namespaces  OR  -xn\n";
 	(*_err) << "    Attach brackets to a namespace statement.\n";
 	(*_err) << endl;
@@ -3209,6 +3212,10 @@ void ASOptions::parseOption(const string
 	{
 		formatter.setAttachClass(true);
 	}
+	else if ( isOption(arg, "xs", "break-structs") )
+	{
+		formatter.setBreakStruct(true);
+	}
 	else if ( isOption(arg, "xk", "attach-extern-c") )
 	{
 		formatter.setAttachExternC(true);
