diff -U 3 -dHrN -- astyle.orig/src/ASFormatter.cpp astyle/src/ASFormatter.cpp
--- astyle.orig/src/ASFormatter.cpp	2014-12-11 22:42:25.000000000 +0800
+++ astyle/src/ASFormatter.cpp	2015-05-21 22:58:24.531580245 +0800
@@ -1895,6 +1895,16 @@
 }
 
 /**
+ * set option to break struct brackets
+ *
+ * @param state        true = break, false = use style default.
+ */
+void ASFormatter::setBreakStruct(bool state)
+{
+	shouldBreakStruct = state;
+}
+
+/**
  * set option to attach extern "C" brackets
  *
  * @param state        true = attach, false = use style default.
@@ -4888,6 +4898,11 @@
 	{
 		return false;
 	}
+	else if (shouldBreakStruct
+	         && isBracketType((*bracketTypeStack)[stackEnd], STRUCT_TYPE))
+	{
+		return true;
+	}
 	else if (shouldAttachInline
 	         && isCStyle()			// for C++ only
 	         && bracketFormatMode != RUN_IN_MODE
diff -U 3 -dHrN -- astyle.orig/src/astyle.h astyle/src/astyle.h
--- astyle.orig/src/astyle.h	2014-12-11 22:42:25.000000000 +0800
+++ astyle/src/astyle.h	2015-05-21 22:58:24.541580210 +0800
@@ -664,6 +664,7 @@
 		void setAttachExternC(bool state);
 		void setAttachNamespace(bool state);
 		void setAttachInline(bool state);
+		void setBreakStruct(bool state);
 		void setBracketFormatMode(BracketMode mode);
 		void setBreakAfterMode(bool state);
 		void setBreakClosingHeaderBracketsMode(bool state);
@@ -883,6 +884,7 @@
 		bool shouldAttachNamespace;
 		bool shouldAttachClass;
 		bool shouldAttachInline;
+		bool shouldBreakStruct;
 		bool isInLineComment;
 		bool isInComment;
 		bool isInCommentStartLine;
diff -U 3 -dHrN -- astyle.orig/src/astyle_main.cpp astyle/src/astyle_main.cpp
--- astyle.orig/src/astyle_main.cpp	2014-12-11 22:42:25.000000000 +0800
+++ astyle/src/astyle_main.cpp	2015-05-21 22:58:24.541580210 +0800
@@ -1667,6 +1667,9 @@
 	cout << "    --attach-extern-c  OR  -xk\n";
 	cout << "    Attach brackets to an extern \"C\" statement.\n";
 	cout << endl;
+	cout << "    --break-structs  OR  -xs\n";
+	cout << "    Break brackets to a namespace statement.\n";
+	cout << endl;
 	cout << "Indentation Options:\n";
 	cout << "--------------------\n";
 	cout << "    --indent-classes  OR  -C\n";
@@ -3013,6 +3016,10 @@
 	{
 		formatter.setAttachClass(true);
 	}
+	else if ( isOption(arg, "xs", "break-structs") )
+	{
+		formatter.setBreakStruct(true);
+	}
 	else if ( isOption(arg, "xk", "attach-extern-c") )
 	{
 		formatter.setAttachExternC(true);
