--- PKGBUILD.orig	2016-08-15 15:00:19.868996640 +0800
+++ PKGBUILD	2016-08-15 15:51:27.322835579 +0800
@@ -21,4 +21,5 @@ source=("https://www.kernel.org/pub/linu
         # standard config files for mkinitcpio ramdisk
         'linux.preset'
+        'hid-apple-swap-fn-ctrl.patch'
         'change-default-console-loglevel.patch')
 
@@ -31,4 +32,5 @@ sha256sums=('5190c3d1209aeda04168145bf50
 validpgpkeys=(
               'ABAF11C65A2970B130ABE3C479BE3E4300411886' # Linus Torvalds
+            'e125693abf49cb210cca6592529fcd98666e166c2874e66ddb3c82188ad89d5b'
               '647F28654894E3BD457199BE38DBBDC86092693E' # Greg Kroah-Hartman
              )
@@ -50,4 +52,7 @@ prepare() {
   patch -p1 -i "${srcdir}/change-default-console-loglevel.patch"
 
+  # add more tuneable parameters for hid-apple module
+  patch -Np1 -i "${srcdir}/hid-apple-swap-fn-ctrl.patch"
+
   if [ "${CARCH}" = "x86_64" ]; then
     cat "${srcdir}/config.x86_64" > ./.config
@@ -77,4 +82,17 @@ prepare() {
   #make oldconfig # using old config from previous kernel version
   # ... or manually edit .config
+  # Enable builtin Intel i915 module
+  ./scripts/config --enable CONFIG_ACPI_BUTTON \
+                   --enable CONFIG_ACPI_VIDEO \
+                   --enable CONFIG_I2C_ALGOBIT \
+                   --enable CONFIG_AGP_INTEL \
+                   --enable CONFIG_INTEL_GTT \
+                   --enable CONFIG_DRM \
+                   --enable CONFIG_DRM_KMS_HELPER \
+                   --enable CONFIG_DRM_I915 \
+                   --enable CONFIG_FB_SYS_FILLRECT \
+                   --enable CONFIG_FB_SYS_COPYAREA \
+                   --enable CONFIG_FB_SYS_IMAGEBLIT \
+                   --enable CONFIG_FB_SYS_FOPS
 
   # rewrite configuration
