--- PKGBUILD.orig	2016-06-09 18:06:37.879715656 -0400
+++ PKGBUILD	2016-06-10 14:05:13.419397628 -0400
@@ -21,4 +21,5 @@ source=("https://www.kernel.org/pub/linu
         # standard config files for mkinitcpio ramdisk
         'linux.preset'
+        'hid-apple-swap-fn-ctrl.patch'
         'change-default-console-loglevel.patch'
         '0001-linux-4.6-rtlwifi-fix-atomic.patch')
@@ -31,4 +32,5 @@ sha256sums=('a93771cd5a8ad27798f22e92405
             'd59014b8f887c6aa9488ef5ff9bc5d4357850a979f3ff90a2999bbe24e5c6e15'
             'f0d90e756f14533ee67afda280500511a62465b4f76adcc5effa95a40045179c'
+            'e125693abf49cb210cca6592529fcd98666e166c2874e66ddb3c82188ad89d5b'
             '1256b241cd477b265a3c2d64bdc19ffe3c9bbcee82ea3994c590c2c76e767d99'
             'ae0d16e81a915fae130125ba9d0b6fd2427e06f50b8b9514abc4029efe61ee98')
@@ -58,4 +60,7 @@ prepare() {
   patch -p1 -i "${srcdir}/0001-linux-4.6-rtlwifi-fix-atomic.patch"
 
+  # add more tuneable parameters for hid-apple module
+  patch -Np1 -i "${srcdir}/hid-apple-swap-fn-ctrl.patch"
+
   if [ "${CARCH}" = "x86_64" ]; then
     cat "${srcdir}/config.x86_64" > ./.config
@@ -85,4 +90,17 @@ prepare() {
   #make oldconfig # using old config from previous kernel version
   # ... or manually edit .config
+  # Enable builtin Intel i915 module
+  ./scripts/config --enable CONFIG_ACPI_BUTTON \
+                   --enable CONFIG_ACPI_VIDEO \
+                   --enable CONFIG_I2C_ALGOBIT \
+                   --enable CONFIG_AGP_INTEL \
+                   --enable CONFIG_INTEL_GTT \
+                   --enable CONFIG_DRM \
+                   --enable CONFIG_DRM_KMS_HELPER \
+                   --enable CONFIG_DRM_I915 \
+                   --enable CONFIG_FB_SYS_FILLRECT \
+                   --enable CONFIG_FB_SYS_COPYAREA \
+                   --enable CONFIG_FB_SYS_IMAGEBLIT \
+                   --enable CONFIG_FB_SYS_FOPS
 
   # rewrite configuration
