--- PKGBUILD.orig	2016-04-14 18:06:41.430191222 -0400
+++ PKGBUILD	2016-04-15 12:03:00.141278196 -0400
@@ -21,4 +21,5 @@ source=("https://www.kernel.org/pub/linu
         # standard config files for mkinitcpio ramdisk
         'linux.preset'
+        'hid-apple-swap-fn-ctrl.patch'
         'change-default-console-loglevel.patch')
 
@@ -50,4 +51,7 @@ prepare() {
   patch -p1 -i "${srcdir}/change-default-console-loglevel.patch"
 
+  # add more tuneable parameters for hid-apple module
+  patch -Np1 -i "${srcdir}/hid-apple-swap-fn-ctrl.patch"
+
   if [ "${CARCH}" = "x86_64" ]; then
     cat "${srcdir}/config.x86_64" > ./.config
@@ -77,4 +81,17 @@ prepare() {
   #make oldconfig # using old config from previous kernel version
   # ... or manually edit .config
+  # Enable builtin Intel i915 module
+  ./scripts/config --enable CONFIG_ACPI_BUTTON \
+                   --enable CONFIG_ACPI_VIDEO \
+                   --enable CONFIG_I2C_ALGOBIT \
+                   --enable CONFIG_AGP_INTEL \
+                   --enable CONFIG_INTEL_GTT \
+                   --enable CONFIG_DRM \
+                   --enable CONFIG_DRM_KMS_HELPER \
+                   --enable CONFIG_DRM_I915 \
+                   --enable CONFIG_FB_SYS_FILLRECT \
+                   --enable CONFIG_FB_SYS_COPYAREA \
+                   --enable CONFIG_FB_SYS_IMAGEBLIT \
+                   --enable CONFIG_FB_SYS_FOPS
 
   # rewrite configuration
